src/darwin/Framework/CHIP/CHIPDevice.mm

```diff
@@ -81,7 +81,7 @@ - (NSString *)openPairingWindowWithPIN:(NSTimeInterval)duration
     uint16_t u16Descriminator = 0;
 
     if (discriminator > 0xfff) {
-        CHIP_LOG_ERROR("Error: Discriminator %ld is too large. Max value %d", discriminator, 0xfff);
+        CHIP_LOG_ERROR("Error: Discriminator %ld is too large. Max value %d", (long) discriminator, 0xfff);
```

Comment:
Why not simply use the portable format specifier for `NSInteger`, `%zd`?

Also, don't we need a bounds check for negative values?  Alternatively, the interface for `openPairingWindowWithPIN` could be modified to receive an NSUInteger (format specifier for that type is `%tu`).

Reply:
Both good suggestions. I'll fix it. Thanks!